cms/djangoapps/contentstore/views/course.py

```diff
@@ -1185,15 +1187,23 @@ def validate(self):
         if len(self.configuration.get('groups', [])) < 1:
             raise GroupConfigurationsValidationError(_("must have at least one group"))
 
-    def generate_id(self, used_ids):
+    @staticmethod
+    def generate_id(used_ids):
         """
         Generate unique id for the group configuration.
         If this id is already used, we generate new one.
         """
-        cid = random.randint(100, 10 ** 12)
+        def random_sql_int():
+            """
+            Generate a random number that can fit in a MySQL INT
+            field, since group and group configuration IDs are stored
+            in the `CourseUserGroupPartitionGroup` table.
+            """
+            return random.randrange(100, 2 ** 31)
```

Comment:
I'd make a named constant for ```2 ** 31``` so its intent can be clear everywhere we use it. It is even possible that we already have one...